File: 0867-transpose-matrix/0867-transpose-matrix.java
class Solution 
{
    public int[][] transpose(int[][] matrix) 
    {
        int r = matrix.length, c = matrix[0].length;
        int ar[][] = new int[c][r];
        for(int i=0; i<r; i++)
            for(int j=0; j<c; j++)
                ar[j][i] = matrix[i][j];
        return ar;
    }
}

File: 0867-transpose-matrix/0867-transpose-matrix.cpp
class Solution 
{
    public:
    vector<vector<int>> transpose(vector<vector<int>>& matrix) 
    {
        int r = matrix.size(), c = matrix[0].size(), j = 0;
        vector<vector<int>> vec(c, vector<int>(r));
        for(vector<int>& log:matrix)
        {
            for(int i=0; i<c; i++)
                vec[i][j] = log[i];
            j++;
        }
        return vec;
    }
};